Shared/Realm.Sync.Shared/Permissions/PermissionChange.cs

```diff
@@ -40,59 +39,114 @@ public class PermissionChange : RealmObject, IPermissionObject
+        public DateTimeOffset CreatedAt { get; private set; } = DateTimeOffset.UtcNow;
 
         /// <inheritdoc />
         [MapTo("updatedAt")]
-        public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
+        public DateTimeOffset UpdatedAt { get; private set; } = DateTimeOffset.UtcNow;
 
         /// <inheritdoc />
         [MapTo("statusCode")]
-        public int? StatusCode { get; set; }
+        public int? StatusCode { get; private set; }
 
         /// <inheritdoc />
         [MapTo("statusMessage")]
-        public string StatusMessage { get; set; }
+        public string StatusMessage { get; private set; }
+
+        /// <inheritdoc />
+        public ManagementObjectStatus ObjectStatus
+        {
+            get
+            {
+                switch (StatusCode)
+                {
+                    case null:
+                        return ManagementObjectStatus.NotProcessed;
+                    case 0:
+                        return ManagementObjectStatus.Success;
+                    default:
+                        return ManagementObjectStatus.Error;
+                }
+            }
+        }
 
         /// <summary>
-        /// Gets or sets the user or users to effect.
+        /// Gets the user or users to effect.
         /// </summary>
         /// <value><c>*</c> to change the permissions for all users.</value>
         [Required]
         [MapTo("userId")]
-        public string UserId { get; set; }
+        public string UserId { get; private set; }
 
         /// <summary>
-        /// Gets or sets the Realm to change permissions for.
+        /// Gets the Realm to change permissions for.
         /// </summary>
         /// <value><c>*</c> to change the permissions of all Realms.</value>
         [Required]
         [MapTo("realmUrl")]
-        public string RealmUrl { get; set; }
+        public string RealmUrl { get; private set; }
 
         /// <summary>
-        /// Gets or sets read access.
+        /// Gets a value indicating whether the user(s) have read access to the specified Realm(s).
         /// </summary>
         /// <value><c>true</c> or <c>false</c> to request this new value. <c>null</c> to keep current value.</value>
         [MapTo("mayRead")]
-        public bool? MayRead { get; set; }
+        public bool? MayRead { get; private set; }
 
         /// <summary>
-        /// Gets or sets write access.
+        /// Gets a value indicating whether the user(s) have write access to the specified Realm(s).
         /// </summary>
         /// <value><c>true</c> or <c>false</c> to request this new value. <c>null</c> to keep current value.</value>
         [MapTo("mayWrite")]
-        public bool? MayWrite { get; set; }
+        public bool? MayWrite { get; private set; }
 
         /// <summary>
-        /// Gets or sets manage access.
+        /// Gets a value indicating whether the user(s) have manage access to the specified Realm(s).
```

Comment:
What does _manage access_ mean? If we have other documentation that explains this in more detail, please link to it.